hls4ml/model/hls_model.py

```diff
@@ -346,6 +346,8 @@ def insert_node(self, node, before=None):
 
         if next_node is not None:
             next_node.inputs[0] = node.outputs[0]
+        else:
+            self.outputs[0] = node.outputs[0]
```

Comment:
I found that before this addition, if an inserted node is the last one in the model, the model outputs are not properly defined anymore